File: frontend/src/api/types/project_member.ts
export type ProjectMemberRole = 'owner' | 'member'

export interface ProjectMemberUser {
  id: number
  username: string
  display_name: string
}

export interface ProjectMember {
  id: number
  project_id: number
  user_id: number
  role: ProjectMemberRole
  user: ProjectMemberUser
}

export interface ProjectMemberListResponse {
  items: ProjectMember[]
  total: number
}

export interface ProjectMemberCreatePayload {
  user_id: number
  role?: ProjectMemberRole
}

export const MEMBER_ROLE_LABELS: Record<ProjectMemberRole, string> = {
  owner: '负责人',
  member: '成员',
}

File: backend/app/schemas/project_member.py
from __future__ import annotations

from typing import List

from pydantic import BaseModel, ConfigDict, Field

from app.models.project_member import ProjectMemberRole


class ProjectMemberUserBrief(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    username: str
    display_name: str


class ProjectMemberResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    project_id: int
    user_id: int
    role: ProjectMemberRole
    user: ProjectMemberUserBrief


class ProjectMemberListResponse(BaseModel):
    items: List[ProjectMemberResponse]
    total: int


class ProjectMemberCreate(BaseModel):
    user_id: int = Field(gt=0)
    role: ProjectMemberRole = ProjectMemberRole.MEMBER
